datafusion/common/src/config.rs

```
@@ -1612,42 +1623,241 @@ impl TableOptions {
+    fn set(&mut self, key: &str, value: &str) -> Result<()> {
+        // Extensions are handled in the public `ConfigOptions::set`
+        let (key, rem) = key.split_once('.').unwrap_or((key, ""));
+        match key {
+            "format" => match self {
+                #[cfg(feature = "parquet")]
+                Self::Parquet { options, .. } => options.set(rem, value),
+                Self::Csv { options, .. } => options.set(rem, value),
+                Self::Json { options, .. } => options.set(rem, value),
+            },
+            _ => _config_err!("Config value \"{key}\" not found on TableOptions"),
         }
+    }
+}
 
-        Ok(ret)
+impl TableFormatOptions {
+    /// Constructs a new instance of `TableOptions` with JSON file type and default settings.
+    ///
+    /// # Returns
+    ///
+    /// A new `TableOptions` instance with default configuration values.
+    pub fn new_json() -> Self {
+        Self::Json {
+            options: Default::default(),
+            extensions: Default::default(),
+        }
     }
 
-    /// Modifies the current `TableOptions` instance with settings from a hash map.
+    /// Constructs a new instance of `TableOptions` with CSV file type and default settings.
+    ///
+    /// # Returns
+    ///
+    /// A new `TableOptions` instance with default configuration values.
+    pub fn new_csv() -> Self {
+        Self::Csv {
+            options: Default::default(),
+            extensions: Default::default(),
+        }
+    }
+
+    /// Constructs a new instance of `TableOptions` with Parquet file type and default settings.
+    ///
+    /// # Returns
+    ///
+    /// A new `TableOptions` instance with default configuration values.
+    #[cfg(feature = "parquet")]
+    pub fn new_parquet() -> Self {
+        Self::Parquet {
+            options: Default::default(),
+            extensions: Default::default(),
+        }
+    }
+
+    pub fn csv_options_or_default(&self) -> CsvOptions {
```

Likewise, these new public methods should have documentation

---

After reading this some more, I don't undertand the usecase for `csv_options_or_default` (and the other similar methods)

Since this is now an enum, the calling code should directly handle the format in the TableFormatOptions -- this code will silently drop any config options if it isn't CSV which seems like it would potentially mask a bug in the caller

I would expect calling code to do something like

```rust
match TableFormatOptions { 
  Self::Csv => {}
  Self::Json => {}
  Self::Parquet => {{
}
```